test/gpu/native/assertOnFailToGpuize.chpl

```diff
@@ -22,14 +22,14 @@ on here.gpus[0] {
   if failureMode == 1 {
```

Comment:
I think I'd remove the individual `.good` files for these cases that now pass and move them to the bottom of this file where we have a bunch of other loops that are expected to work. Then renumber the failure modes (e.g. move `assertOnFailToGpuize.3.good` through `assertOnFailToGpuize.5.good` to `assertOnFailToGpuize.1.good` to `assertOnFailToGpuize.3.good`, get rid of modes 6&7 and then have 6 be the "cases we expect to pass" mode, and then update the `.compopt` file to iterate from 1 to 6.)

It's kind of a pain, but I also kind of like having this test as kind of a self-documenting list of the known failure modes that `assertOnGpu()` can capture, and it's kind of misleading to have something that's meant to pass under `if failureMode == x { ... }`.

Reply:
I updated the test do do this.